File: 프로그래머스/0/120913. 잘라서 배열로 저장하기/잘라서 배열로 저장하기.java
import java.util.*;
import java.util.stream.*;
class Solution {
    public String[] solution(String my_str, int n) {
        List<String> ret = new ArrayList<>();
        for(int i = 0; i < my_str.length(); i += n){
            int end = Math.min(i + n, my_str.length());
            ret.add(my_str.substring(i, end));
        }
        return ret.stream().toArray(String[]::new);
    }
}

File: 프로그래머스/0/120913. 잘라서 배열로 저장하기/잘라서 배열로 저장하기.py
def solution(my_str, n):
    temp = ''
    ret = []
    for c in my_str:
        if len(temp) == n:
            ret.append(temp)
            temp = c
        else: 
            temp += c
    if temp: 
        ret.append(temp)
    return ret